histomicstk/features/ComputeHaralickFeatures.py

```diff
@@ -0,0 +1,309 @@
+import numpy as np
+import pandas as pd
+from skimage.measure import regionprops
+from graycomatrixext import graycomatrixext
+
+
+def ComputeHaralickFeatures(im_label, im_intensity, offsets=None, num_levels=8,
+                            gray_limits=[0, 255], rprops=None):
+    """
+    Calculates 26 Haralick features from an intensity image with the labels.
+
+    Parameters
+    ----------
+    im_label : array_like
+        A labeled mask image wherein intensity of a pixel is the ID of the
+        object it belongs to. Non-zero values are considered to be foreground
+        objects.
+
+    im_intensity : array_like
+        Intensity image
+
+    offsets : array_like, optional
+        A (num_offsets, num_image_dims) array of offset vectors
+        specifying the distance between the pixel-of-interest and
+        its neighbor. Note that the first dimension corresponds to
+        the rows.
+
+        Because this offset is often expressed as an angle, the
+        following table lists the offset values that specify common
+        angles for a 2D image, given the pixel distance D.
+
+        AngleXY  |  OFFSET
+        -------  |  ------
+        0        |  [0 D]
+        45       |  [-D D]
+        90       |  [-D 0]
+        135      |  [-D -D]
+
+        Default
+            [1] for 1D,
+            [[1, 0], [0, 1]] for 2D,
+            [[1, 0, 0], [0, 1, 0], [0, 0, 1] for 3D and so on
+
+    num_levels : unsigned int, optional
+        An integer specifying the number of gray levels For example, if
+        `num_levels` is 8,  the intensity values of the input image are
+        scaled so they are integers between 1 and 8.  The number of gray
+        levels determines the size of the gray-level co-occurrence matrix.
+
+        Default: 8 for numeric image, 2 for binary/logical image
+
+    gray_limits : array_like, optional
+        A two-element array specifying the desired input intensity range.
+        Intensity values in the input image will be clipped into this range.
+
+        Default: [0, 1] for binary/logical image, [0, 255] for numeric image
+
+    Returns
+    -------
+    fdata: pandas.DataFrame
+        A pandas dataframe containing the haralick features.
+
```

Comment:
Add `Notes` in doc string to describe the features that this function computes. See [here](https://github.com/DigitalSlideArchive/HistomicsTK/blob/master/histomicstk/features/ComputeIntensityFeatures.py#L37-L77).